src/docgen.cpp

```diff
@@ -5,13 +5,14 @@
 #include <string>
 #include <iostream>
 #include <fstream>
+#include <memory>
 #include <nlohmann/json.hpp>
 #include "exceptions.hpp"
 
 /* Options */
-static char **files_src_paths;
-static std::istream *parsed_src = NULL;
-static std::ostream *parsed_dst = NULL;
+static const char * const *files_src_paths;
+static std::shared_ptr<std::istream> parsed_src(nullptr, [](std::istream *p){});
```

Comment:
deleting nullptr is well-defined and will do nothing. That second argument is actually not needed. Also nice for using lambda function I don't know why I didn't think of that xD

Reply:
Ah right, good point!